### DISCUSSION TITLE
JDBC가 무엇인가? JDBC의 구현체는 어디에 있고 어떻게 사용할 수 있는가?

### DISCUSSION THREAD

**Discussion by seunghye218** [2024-woowacourse-study/level-interview#76] [Java]
어떻게 벤더사(Oracle DB, MySQL, MariaDB, PostgreSQL, ...)와 상관 없이 자바 코드를 작성할 수 있는가?

**Comment by alstn113** [comment]
JDBC(Java Database Connectivity)는 Java 기반 애플리케이션의 데이터를 데이터베이스에 저장 및 업데이트하거나, 데이터베이스에 저장된 데이터를 Java에서 사용할 수 있도록 하는 자바 API이다.
JDBC는 Java 애플리케이션에서 데이터베이스에 접근하기 위해 JDBC API를 사용하여 데이터베이스에 연동할 수 있으며, 데이터베이스에서 자료를 쿼리(Query)하거나 업데이트하는 방법을 제공한다.

JDBC는 3가지 기능을 표준 인터페이스로 정의하여 제공한다.
1. java.sql.Connection - 연결
2. java.sql.Statement - SQL을 담은 내용
3. java.sql.ResultSet - SQL 요청 응답

JDBC의 동작 흐름은 다음과 같다.
Java 애플리케이션 -> JDBC API -> JDBC 드라이버 -> 데이터베이스

JDBC드라이버는 데이터베이스와의 통신을 담당하는 인터페이스이다. 
Oracle, MS SQL, MySQL 등과 같은 데이터베이스에 알맞은 JDBC 드라이버를 구현하여 제공한다.
JDBC 드라이버의 구현체를 이용해서 특정 벤더의 데이터베이스에 접근할 수 있다.

**Comment by GIVEN53** [comment]
자바 진영에서 데이터베이스에 접근할 수 있도록 하는 API이다.
데이터베이스 접근을 표준화했기 때문에 특정 벤더사에 종속적이지 않으며 데이터베이스 변경 또한 용이하다.

![image](https://github.com/2024-woowacourse-study/level-interview/assets/101033262/9c95f2b0-bcd3-46d0-b7db-16e7345123f7)
